src/main/java/com/board/config/boardconfiggui/common/Utils.java

```diff
@@ -57,11 +57,18 @@ public static boolean saveData(String folderPath){
      * @param title     The title of the alert dialog.
      * @param content   The content text to be displayed in the alert dialog.
      */
-    private static void alertDialog(Alert.AlertType alertType, String title, String content){
+    public static void alertDialog(Alert.AlertType alertType, String title, String content){
         Alert alert = new Alert(alertType);
         alert.setTitle(title);
         alert.setHeaderText(null);
         alert.setContentText(content);
         alert.showAndWait();
     }
+
+    public static boolean validateXmlFolder(String xmlFolderPath) {
+        File hardwareConfigFile = new File(xmlFolderPath + "/hardware_configuration.xml");
```

Comment:
Use filename from constants and follow this pattern to get the file instance new File(directory, filename)